### PR TITLE
Add support to build docker image with eclipselink

### PR DESCRIPTION
**Pull request by MonkeyCanCode**
# Description

As https://github.com/apache/polaris/pull/114 added support for building the project with eclipselink via optional flag, this PR is add the same support via optional command line argument  to `run.sh` when using mini-deployment locally. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Default:
```
yong@DESKTOP:~/polaris(optional_eclipselink)$ bash run.sh
Building Kind Registry...
...
Creating cluster "kind" ...
...
Building polaris image with ECLIPSELINK=false...
...
Applying Kubernetes manifests...
namespace/polaris created
deployment.apps/polaris-deployment created
service/polaris-service created
```

With optional command line argument:
```
yong@DESKTOP:~/polaris(optional_eclipselink)$ bash run.sh -e true
Building Kind Registry...
...
Creating cluster "kind" ...
...
Building polaris image with ECLIPSELINK=true...
...
Applying Kubernetes manifests...
namespace/polaris created
deployment.apps/polaris-deployment created
service/polaris-service created
```

Helper:
```
xxx@DESKTOP:~/polaris(optional_eclipselink)$ bash run.sh -h
Usage: run.sh [-e true|false] [-h]
  -e    Set the ECLIPSELINK flag (default: false)
  -h    Display this help message
```

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -21,20 +21,51 @@
...
+# Deploy the registry
 echo "Building Kind Registry..."
 sh ./kind-registry.sh
 
-# build and deploy the server image
-echo "Building polaris image..."
-docker build -t localhost:5001/polaris -f Dockerfile .
+# Build and deploy the server image
+echo "Building polaris image with ECLIPSELINK=$ECLIPSELINK..."
+docker build -t localhost:5001/polaris --build-arg ECLIPSELINK=$ECLIPSELINK -f Dockerfile .
```
Comment (eric-maynard) [inline run.sh:63]: Is this always true?

File: run.sh [hunk 2]
```diff
@@ -21,20 +21,51 @@
...
+ECLIPSELINK="false"  # Default value
 
-# deploy the registry
+# Function to display usage information
+usage() {
+  echo "Usage: $0 [-e true|false] [-h]"
+  echo "  -e    Set the ECLIPSELINK flag (default: false)"
+  echo "  -h    Display this help message"
+  exit 1
+}
```
Comment (eric-maynard) [inline run.sh:33]: This change is one I'm a little unsure about. Things like metastore manager and callcontext resolver are designed to be pluggable and discoverable by DropWizard. 

I'm worried about the precedent this sets and whether we would want to add all these as myriad options here. The script might get quite complicated in order to cover every combination.

File: run.sh (hunk 1 above)
Comment (MonkeyCanCode) [reply run.sh:63]: Default is false on line 25 which matches the behavior of current gradle build.

File: run.sh (hunk 2 above)
Comment (MonkeyCanCode) [reply run.sh:33]: With this concern, should we move those into a config file instead?

Comment (eric-maynard) [reply run.sh:33]: Or maybe just a safety valve type flag that allows you to pass an arbitrary build arg down to `docker build`?

Comment (MonkeyCanCode) [reply run.sh:33]: K, let me think about it tonight then submit another PR to address those tonight.

Comment (MonkeyCanCode) [reply run.sh]: @eric-maynard updated with arbitrary build arg support. Please take another look when you get a chance.

File: run.sh [hunk 3]
```diff
@@ -21,18 +21,57 @@
...
+shift $((OPTIND-1))
+
+# Deploy the registry
 echo "Building Kind Registry..."
 sh ./kind-registry.sh
 
-# build and deploy the server image
+# Check if BUILD_ARGS is not empty and print the build arguments
+if [[ -n "$BUILD_ARGS" ]]; then
+  echo "Building polaris image with build arguments:$BUILD_ARGS"
```
Comment (eric-maynard) [inline run.sh:62]: nit: space missing after `:`

File: run.sh [hunk 4]
```diff
@@ -21,18 +21,57 @@
...
+docker build -t localhost:5001/polaris $BUILD_ARGS -f Dockerfile .
 echo "Pushing polaris image..."
 docker push localhost:5001/polaris
 echo "Loading polaris image to kind..."
 kind load docker-image localhost:5001/polaris:latest
 
 echo "Applying kubernetes manifests..."
 kubectl delete -f k8/deployment.yaml --ignore-not-found
-kubectl apply -f k8/deployment.yaml
+kubectl apply -f k8/deployment.yaml
```
Comment (eric-maynard) [inline run.sh:77]: Missing whitespace at end of file

**Review by eric-maynard — APPROVED** [review]
LGTM with some comments on style

File: run.sh (hunk 3 above)
Comment (MonkeyCanCode) [reply run.sh:62]: Took care.

File: run.sh (hunk 4 above)
Comment (MonkeyCanCode) [reply run.sh:77]: Took care.

**Comment by MonkeyCanCode** [comment]
Anything else needed on this PR?

**Review by RussellSpitzer — APPROVED** [review]

**Comment by RussellSpitzer** [comment]
The license check is currently failing, I don't think this should be happening, can you try rebasing and try again?

**Comment by MonkeyCanCode** [comment]
> The license check is currently failing, I don't think this should be happening, can you try rebasing and try again?

yes. Just noticed it. Was fine earlier. Had being a while since I submitted this PR. The PR had being rebased.

**Comment by adutra** [comment]
> > The license check is currently failing, I don't think this should be happening, can you try rebasing and try again?
> 
> yes. Just noticed it. Was fine earlier. Had being a while since I submitted this PR. The PR had being rebased.

Probably fixed by #266 or #271.

File: run.sh (hunk 2 above)
Comment (MonkeyCanCode) [reply run.sh]: actually lets have it in a diff PR. The current Dockerfile doesn't support eclipseLinkDeps flag. I will add this in and update the doc for both optional options. What do u think?

@flyrain

Comment (flyrain) [reply run.sh]: I'd recommend to add an example like  `./run.sh -b ECLIPSELINK=true`, as it is probably most commonly used.

**Review by flyrain — APPROVED** [review]
+1

**Comment by flyrain** [comment]
Thanks @MonkeyCanCode for the PR. Thanks @RussellSpitzer @eric-maynard @collado-mike for the review.

**Pull request merged**